### ISSUE TITLE
chakra-ui Checkbox and Radio components require double click, but non-issue for NextJS

### ISSUE THREAD

**Issue by chrisbull** [blitz-js/blitz#1838]
Please keep open for others to find until this is resolved.

### What is the problem?

See comments from #1600 

The issue is that when using `chakra-ui` and `blitz`, blitz has invoked experimental react flags that are causing Checkbox and Radio to not work correctly and must be clicked twice, and quickly. 

Issues do not pertain to any other libraries (such as react-final-form or others)

### Steps to Reproduce

1. Create a fresh Blitz app `blitz new my-blitz-app`
2. Install chakra-ui `blitz install chakra-ui`
3. On the login page, add a `<Checkbox>Rememer Me</Checkbox>` chakra-ui component
4. run

### Versions
0.29.7  and 0.30.0-canary.4 

### Other
If you install a new app using `create-next-app` and use the exact same code, as well as install react@experimental there is no issues with chakra-ui. So something blitz is doing is affecting this.

**Comment by anteprimorac** [comment]
The fix for this https://github.com/chakra-ui/chakra-ui/pull/3165 has already been merged into chakra. Still waiting for a new version to be published on npm.

**Comment by anteprimorac** [comment]
There is a new version of chakra published yesterday. Can you make sure that you are using the latest version `1.2.1`?

**Comment by osirvent** [comment]
> There is a new version of chakra published yesterday. Can you make sure that you are using the latest version `1.2.1`?

@Zeko369 is the chakra-ui recipe maintainer

**Comment by anteprimorac** [comment]
@chrisbull here is a working example https://codesandbox.io/s/concurrent-mode-fixed-fyuiz

**Comment by lcswillems** [comment]
It seems the issue is coming from Blitz.

**Comment by flybayer** [comment]
@lcswillems Blitz doesn't change anything with react rendering, so it's unlikely.

Perhaps it could have to do with which react experimental version you are using? Try updating to the latest react experimental versions. (`yarn add react@experimental react-dom@experimental`)

**Comment by flybayer** [comment]
I just noticed @anteprimorac's sandbox is using create react app instead of next.js. So you should test this with a next.js or [blitz codesandbox](https://sandbox.blitzjs.com).

**Comment by anteprimorac** [comment]
here is a working example created with blitz codesandbox template https://codesandbox.io/s/exciting-noether-x6er0

**Comment by lcswillems** [comment]
Here is a version with the bug: https://codesandbox.io/s/bold-sky-gk21l

It seems it is related to `FormControl`

**Comment by lcswillems** [comment]
The issue seems to come from Blitz.

With create-react-app, there is no bug: https://codesandbox.io/s/unruffled-lovelace-fcjor
With Blitz, there is the bug: https://codesandbox.io/s/bold-sky-gk21l

**Comment by anteprimorac** [comment]
The main issue is with new react concurrent mode https://github.com/facebook/react/issues/18591. Blitz use it by default, but you can opt-in for legacy react mode.

Here is the codesandbox example without blitz that shows described error https://codesandbox.io/s/concurrent-mode-fyuiz. I will post a link to PR that fixes this issue in chakra once I create it.

**Comment by lcswillems** [comment]
Okay, I see, the version of create-react-app I used was not using concurrent mode!

How can I opt out?

**Comment by anteprimorac** [comment]
By setting `experimental.reactMode` to `legacy` in blitz.config.js:
```
module.exports = {
  // ...
  experimental: {
    reactMode: 'legacy'
  },
  // ...
};
```

**Comment by lcswillems** [comment]
Thank you!

**Comment by flybayer** [comment]
Concurrent mode is now fixed for me as of react/react-dom `0.0.0-experimental-1fafac002` with react-final-form, so very likely Chakra is fixed too!
